core/modules/layout/layout.module

```diff
@@ -544,9 +544,131 @@ function layout_menu() {
+            $access = array_intersect_key($bundle_info['admin'], backdrop_map_assoc(array('access callback', 'access arguments')));
+            $access += array(
+              'access callback' => 'user_access',
+              'access arguments' => array('administer layouts'),
+            );
+
+            $items["$path/layouts"] = array(
+              'title' => 'Manage layouts',
+              'page callback' => 'backdrop_get_form',
+              'page arguments' => array('layout_entity_admin_form', $entity_type, $bundle_arg),
+              'type' => MENU_LOCAL_TASK,
+              'weight' => 2,
+              'file' => 'layout.entity.admin.inc',
+            ) + $access;
+            $items["$path/layouts/add-bundle-layout"] = array(
+              'title' => "Add layout",
+              'title callback' => 'layout_entity_menu_title',
+              'title arguments' => array($entity_type, $bundle_arg),
+              'page callback' => 'backdrop_get_form',
+              'page arguments' => array('layout_entity_admin_add_form', $entity_type, $bundle_arg),
+              'file' => 'layout.entity.admin.inc',
+              'type' => MENU_LOCAL_ACTION,
+            ) + $access;
+            $items["$path/layouts/add-entity-layout"] = array(
+              'title' => "Add layout",
+              'title callback' => 'layout_entity_menu_title',
+              'title arguments' => array($entity_type, $bundle_arg, 'create_for_all_bundles'),
+              'page callback' => 'backdrop_get_form',
+              'page arguments' => array('layout_entity_admin_add_form', $entity_type, $bundle_arg, TRUE),
+              'file' => 'layout.entity.admin.inc',
+              'type' => MENU_LOCAL_ACTION,
+            ) + $access;
+          }
+        }
+      }
+    }
+  }
+
   return $items;
 }
 
+/**
+ * Title callback: Returns the title of an entity layout page.
+ *
+ * @see layout_menu()
+ */
+function layout_entity_menu_title($entity_type, $bundle_arg = NULL, $create_for_all_bundles = NULL) {
+  $entity_info = entity_get_info($entity_type);
+  $entity_name = ($entity_type == 'node') ? t('content') : $entity_info['label'];
+  $bundle_name = '';
+
+  // This entity has only a single bundle.
+  if (!$bundle_arg) {
+    return t('Create a layout for @entity types', array('@entity' => $entity_name));
+  }
+  // This is the action link to create a layout for a particular bundle.
+  elseif (!$create_for_all_bundles) {
+    if (is_object($bundle_arg) && property_exists($bundle_arg, 'name')) {
+      $bundle_name = $bundle_arg->name;
+    }
+    elseif (is_object($bundle_arg) && property_exists($bundle_arg, 'label')) {
+      $bundle_name = $bundle_arg->label;
+    }
+    elseif (is_string($bundle_arg)) {
+      if (isset($entity_info['label'])) {
+        $bundle_name = $entity_info['label'];
+      }
+    }
+    return t('Create a layout for [@bundle_name] @entities', array('@bundle_name' => $bundle_name, '@entities' => $entity_name));
```

Comment:
Can we please use italics here instead of brackets around the bundle name?
```suggestion
    return t('Create a layout for %bundle_name @entities', array('%bundle_name' => $bundle_name, '@entities' => $entity_name));
```